src/synthesizer/typescript/mod.rs

```diff
@@ -394,8 +413,14 @@ fn emit_resource_metadata(
     match metadata {
         ResourceIr::Object(_, entries) => {
             for (name, value) in entries {
-                output.text(format!("{name}: "));
-                emit_resource_ir(context, &output, value, Some(",\n"));
+                // Wrap the prop name with quotes ('') to support special characters in it
+                output.text(format!("'{name}': "));
```

Comment:
I personally preferred the non quoted version (kept going back and forth on the initial implementation):\ 

Do we know the characters that aren't allowed, and we can escape them (or is that too much logic?) In the end, I want folks to not have to rewrite a bunch, and I feel most would remove the quotes.

Reply:
@iph I couldn't find what chars are allowed or not allowed in the Javascript properties without quotes. This (https://mathiasbynens.be/notes/javascript-properties) doc might help.

Reply:
fair enough. I think it's not worth complicating -- so I'm cool with this. Running workflows